src/thing/Portal.cpp

```diff
@@ -50,13 +57,44 @@ void Portal::Start() {
           }
         }
       }
+      function scan() {
+        document.getElementById('scanning').style.display='inline';
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", "/wifi/scan", true);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState==4 && xhr.status==200) {
+           load_networks(JSON.parse(xhr.responseText));
+           document.getElementById('scanning').style.display='none';
+          }
+        }
+        xhr.send();
+      }
+      function load_networks(networks) {
+        select = document.getElementById('networks');
+        select.innerHtml = '';
+        networks.sort(function(a, b) {
+          return b.rssi - a.rssi;
+        });
+        networks.forEach(function(network) {
+          option = document.createElement('option');
+          option.value = network.ssid;
+          option.text = network.ssid + ' (' + network.rssi +  ')';
+          select.add(option);
+        });
+        select.style.display='inline';
```

Comment:
I am not sure I follow what you are proposing. It may just be beyond my CSS knowledge. The UI is very much in a "working but unstyled" state. I was planning on going back at the end and making things look good, add some CSS to try to match the feel of the firebase console, stuff like that but if someone wanted to jump in and start styling it now I wouldn't complain :)